infrastructure/controller/pkg/endpoints/cluster_apply.go

```diff
@@ -0,0 +1,97 @@
+package endpoints
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/airyhq/airy/lib/go/config"
+	"github.com/airyhq/airy/lib/go/k8s"
+	"github.com/airyhq/airy/lib/go/payloads"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+type ClusterApply struct {
+	clientSet *kubernetes.Clientset
+	namespace string
+}
+
+func (s *ClusterApply) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+
+	var conf config.AiryConf
+	namespace := "default"
+	responseConfigComponents := payloads.ConfigApplyResponsePayload{}
+
+	err = json.Unmarshal(body, &conf)
```

Comment:
As mentioned I would suggest that we nest the config inside a `data` field. This will make the API more maintainable as we can add more options to the request body later when needed.